internal/model/plugin.go

```diff
@@ -7,13 +7,19 @@ import (
 	mattermostModel "github.com/mattermost/mattermost-server/model"
 )
 
+// PluginSignatures represents Mattermost plugin signatures
+type PluginSignatures struct {
+	Signature     string `json:"signature"`
+	PublicKeyHash string `json:"public_key_hash"`
+}
+
 // Plugin represents a Mattermost plugin in the marketplace.
 type Plugin struct {
-	HomepageURL       string                    `json:"homepage_url"`
-	IconData          string                    `json:"icon_data"`
-	DownloadURL       string                    `json:"download_url"`
-	DownloadSignature []byte                    `json:"download_signature"`
-	Manifest          *mattermostModel.Manifest `json:"manifest"`
+	HomepageURL string                    `json:"homepage_url"`
+	IconData    string                    `json:"icon_data"`
+	DownloadURL string                    `json:"download_url"`
+	Manifest    *mattermostModel.Manifest `json:"manifest"`
+	Signatures  []*PluginSignatures       `json:"signatures"`
```

Comment:
Should we converge on `[]PluginSignatures` in this case?

Reply:
Yep, I was going for consistency here. 
In this concrete case either will work - copying 4 arguments vs dereferencing 2. 
But in general I don't think it's a good idea using `[]struct` everywhere.
1/5 that we should stick to `[]*PluginSignatures`

Reply:
i wish we had used non-pointers in our other codebase :(, i miss just using `[]struct` :( for simplicity.

Reply:
Agree we should reach for consistency within a repository at a minimum. I had been under the impression we were already using vanilla structs, but looks like I'm wrong, so no changes required here.

As I understand, it's hard to reason about Go's handling of structs passed by value, since it may choose to promote to the heap anyway or it might take optimization steps to avoid even the expense of copying. So if we had to make a decision today, I'd say `[]struct` and trust the Go compiler. 

I'd be open to "fixing" this repository as such and starting a new convention, since I don't think `[]*struct` is a coding style thing that we have to enforce.